tests/test_headful.py

```diff
@@ -0,0 +1,110 @@
+"""
+Tests relating to headful mode
+"""
+import os
+import platform
+import pytest
+import tempfile
+import shutil
+
+from pyppeteer import launch
+from syncer import sync
+
+EXTENSION_PATH = os.path.join(os.path.dirname(os.path.realpath(__file__)), "assets", "simple-extension")
+
+
+class TestHeadful:
+    @pytest.fixture
+    def extension_options(self, default_browser_options):
+        default_browser_options["headless"] = False
+        default_browser_options["args"] += [
+            '--disable-extensions-except=' + EXTENSION_PATH,
+            '--load-extension=' + EXTENSION_PATH
+        ]
+        return default_browser_options
+
+    @pytest.fixture
+    def headful_options(self, default_browser_options):
+        default_browser_options["headless"] = False
+        return default_browser_options
+
+    @sync
+    async def test_background_page_target_type_should_be_available(self, extension_options):
+        browser_with_extension = await launch(**extension_options)
```

Comment:
Do we need docstrings for each test to understand more precisely what test was intended to do?

Reply:
I think the test names are self explanatory. Even Puppeteer tests don't have comments.

Reply:
IMO as long as the test name is fairly self-explanatory this is fine.